### DISCUSSION TITLE
How to using content block in conditions

### DISCUSSION THREAD

**Discussion by MrBoriska** [typst/typst#4339] [Q&A] [answered]
code

```
#let part_count = counter("parts")
#let total_part = context[#part_count.final().at(0)]
```

and try condition with total_part variable:

`#if total_part > 0 [#total_part parts]`

but give error:

```
error: cannot compare content and integer
    ┌─ parts/intro.typ:109:4
    │
109 │ #if total_part > 0 [#total_part parts]
```

how to using content block in conditions? #total_part shows as 0, but in condition #total_part not equal [0] or 0 or "0".

**Comment by l0wx** [comment, marked as answer]
Cf. https://github.com/typst/typst/discussions/3683#discussioncomment-8807426

**Comment by l0wx** [comment]
Cf. https://github.com/typst/typst/discussions/3895#discussioncomment-9037801

**Comment by Mc-Zen** [comment]
Hi @MrBoriska , 
getting counter values requires context but this will totally work:
```typ
#context {
  let total_part = part_count.final().at(0)
  if total_part >= 0 [#total_part parts]
}
```
The difference is, that the counter's value is not converted to content (until the second line, in `[#total_part parts]`), achieved by wrapping all of it in one `context` block.
